src/psyclone/domain/lfric/lfric_stencils.py

```diff
@@ -0,0 +1,649 @@
+        return symtab.find_or_create_array(root_name, num_dimensions,
+                                           ScalarType.Intrinsic.INTEGER,
+                                           tag=unique)
+
+    @staticmethod
+    def dofmap_size_symbol(symtab, arg):
+        '''
+        Create a valid symbol for the size (in cells) of a stencil
+        dofmap in the PSy layer.
+
+        :param symtab: symbol table that will contain (or already contains) \
+            the symbol with this name.
+        :type symtab: :py:class:`psyclone.psyir.symbols.SymbolTable`
+        :param arg: the kernel argument with which the stencil is associated.
+        :type arg: :py:class:`psyclone.dynamo0p3.DynKernelArgument`
+
+        :returns: a symbol for the stencil size.
+        :rtype: :py:class:`psyclone.psyir.symbols.Symbol`
+
+        '''
+        root_name = arg.name + "_stencil_size"
+        unique = LFRicStencils.stencil_unique_str(arg, "size")
+        if arg.descriptor.stencil['type'] == "cross2d":
+            num_dimensions = 2
+        else:
+            num_dimensions = 1
+        return symtab.find_or_create_array(root_name, num_dimensions,
+                                           ScalarType.Intrinsic.INTEGER,
+                                           tag=unique)
+
+    @staticmethod
+    def max_branch_length_name(symtab, arg):
+        '''
+        Create a valid unique name for the maximum length of a stencil branch
+        (in cells) of a 2D stencil dofmap in the PSy layer. This is required
+        in the kernels for defining the maximum possible length of one of the
+        dofmap array dimensions.
+
+        :param symtab: symbol table that will contain (or already contains) \
+            the symbol with this name.
+        :type symtab: :py:class:`psyclone.psyir.symbols.SymbolTable`
+        :param arg: the kernel argument with which the stencil is associated.
+        :type arg: :py:class:`psyclone.dynamo0p3.DynKernelArgument`
+
+        :returns: a Fortran variable name for the max stencil branch length.
+        :rtype: str
+        '''
+        root_name = arg.name + "_max_branch_length"
+        unique = LFRicStencils.stencil_unique_str(arg, "length")
+        return symtab.find_or_create_integer_symbol(root_name, tag=unique).name
+
+    def _unique_max_branch_length_vars(self):
+        '''
+        :returns: list of all the unique max stencil extent argument names in
+                  this kernel call for cross2d stencils.
+        :rtype: list of str
+        '''
+        names = []
+        for arg in self._kern_args:
+            if arg.descriptor.stencil['type'] == "cross2d":
+                names.append(arg.name + "_max_branch_length")
+
+        return names
+
+    def _declare_unique_max_branch_length_vars(self, parent):
+        '''
+        Declare all unique max branch length arguments as integers with intent
+        in and add the declaration as a child of the parent argument passed
```

Comment:
I would suggest single quotations around `'in'` here to mark it is code (intent).